dali/base/dasds.cpp

```diff
@@ -2809,6 +2809,8 @@ class CNodeSubscriptionManager : public CSimpleInterface, implements INodeSubscr
     // ISubscriptionManager impl.
     virtual void add(ISubscription *sub, SubscriptionId id)
     {
+        CHECKEDDALIREADLOCKBLOCK(owner.dataRWLock, readWriteTimeout);
+        CHECKEDCRITICALBLOCK(owner.treeRegCrit, fakeCritTimeout);
         CriticalBlock b(lock);
```

Comment:
It would be clearer if lock was named to describe what it was it was locking.  It it just the subscription lists?